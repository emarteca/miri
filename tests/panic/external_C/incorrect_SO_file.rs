//@rustc-env: RUST_BACKTRACE=0
//@only-target-linux
//@only-on-host
//@compile-flags: -Zmiri-external_c_so_file=tests/external_C/badpath.so
//@normalize-stderr-test: "note: rustc.*running on.*" -> ""

extern "C" {
    fn printer();
}

fn main() {
    unsafe {
        // calling a function from a shared object file that doesn't exist
        printer();
    }
}
