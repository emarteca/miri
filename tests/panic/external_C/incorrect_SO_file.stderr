thread 'main' panicked at '-Zmiri-external_c_so_file path tests/external_C/badpath.so does not exist', src/bin/miri.rs:LL:CC
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md



note: compiler flags: -Z ui-testing -Z miri-external_c_so_file=tests/external_C/badpath.so

query stack during panic:
end of query stack
